### PR TITLE
[add] add emoji_InputFormatter support our developer use emoji

### PR DESCRIPTION
**Pull request by Linsixu**
## Description

When the input box supports emoticons, when we have text and emoticons in the input box at the same time, there will be abnormal text length and cursor. For example, maxLength = 15, when there is an emoticon in the input box, the length of the text in the input box can reach 16, and then the cursor will automatically move one bit.

中文描述：
输入框支持表情时候，当我们输入框中同时存在文字与表情符号时候，会出现文字长度与光标异常。例如，maxLength= 15，当输入框中有表情符号时候，输入框的文案长度可以达到16，然后光标会自动迁移一位。

## Related Issues
https://github.com/flutter/flutter/issues/55670
## Tests

I added the following tests:

Normal characters do not contain expressions, normal
Normal characters contain expressions, normal
The maximum length can be just right

### CODE REVIEW

File: packages/flutter/lib/src/services/text_formatter.dart [hunk 1]
```diff
@@ -122,6 +122,70 @@ class BlacklistingTextInputFormatter extends TextInputFormatter {
   static final BlacklistingTextInputFormatter singleLineFormatter
       = BlacklistingTextInputFormatter(RegExp(r'\n'));
 }
+///Used to support the problem of limiting the maximum length when there is an emoji in the input box
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/text_formatter.dart:125]: This is not a proper doc comment. It should describe what this class does and why one should use it.

**Review by goderbauer — COMMENTED** [review]
I am not sure what the purpose of this PR is. Looks like you could easily publish this as an external package on pub.

Also, when you opened the PR, there was a checklist in the PR description. Can you please fill out that checklist and ensure that you've done all the items on that list? Among other things, this PR is missing tests, its missing a signed CLA, the analyzer is failing, etc.

**Comment by justinmc** [comment]
I have a PR in progress that fixes the length problem (https://github.com/flutter/flutter/pull/53381), but I'm not sure if it does everything that this PR does.

**Comment by goderbauer** [comment]
I am going to close this one since there hasn't been any follow-up.

File: packages/flutter/lib/src/services/text_formatter.dart (hunk 1 above)
Comment (Linsixu) [reply packages/flutter/lib/src/services/text_formatter.dart:125]: Thank to your suggestions. Relevant description has been added

**Comment by Linsixu** [comment]
> I have a PR in progress that fixes the length problem (#53381), but I'm not sure if it solves everything that this PR is trying to solve.

Thank to your suggestions. bug it do not work for my question.

**Pull request closed without merging**